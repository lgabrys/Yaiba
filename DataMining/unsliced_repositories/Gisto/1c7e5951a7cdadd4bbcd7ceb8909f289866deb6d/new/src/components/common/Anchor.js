import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { baseAppColor } from 'constants/colors';

const Link = styled.a`
  cursor: pointer;
  color: ${baseAppColor};
  text-decoration: none;
`;

const Anchor = ({
  href, onClick, children, download
}) => (
  <Link href={ href }
        download={ download }
        onClick={ onClick }>
    { children }
  </Link>
);

Anchor.propTypes = {
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
  download: PropTypes.string
};

export default Anchor;
