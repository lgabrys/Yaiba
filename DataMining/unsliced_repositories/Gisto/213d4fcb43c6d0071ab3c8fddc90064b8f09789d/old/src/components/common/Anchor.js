import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { baseAppColor } from 'constants/colors';

const Link = styled.a`
  cursor: pointer;
  margin: 0 5px 0 0;
  color: ${baseAppColor};
  text-decoration: none;
`;

const Anchor = ({
  key, href, onClick, children 
}) => (
  <Link href={ href }
        key={ key }

        onClick={ onClick }>
    { children }
  </Link>
);

Anchor.propTypes = {
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
  key: PropTypes.any
};

export default Anchor;
