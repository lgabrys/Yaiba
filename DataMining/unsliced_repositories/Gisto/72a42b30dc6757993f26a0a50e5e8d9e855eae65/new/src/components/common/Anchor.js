import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { baseAppColor, headerBgLightest } from 'constants/colors';

const Link = styled.a`
  cursor: pointer;
  color: ${(props) => props.color ? props.color : baseAppColor};
  text-decoration: none;

  :hover {
    color: ${headerBgLightest};
  }
`;

const Anchor = ({
  href, onClick, children, download, color, className
}) => (
  <Link href={ href }
        className={ className }
        color={ color }
        download={ download }
        onClick={ onClick }>
    { children }
  </Link>
);

Anchor.propTypes = {
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
  download: PropTypes.string,
  color: PropTypes.string,
  className: PropTypes.string
};

export default Anchor;
