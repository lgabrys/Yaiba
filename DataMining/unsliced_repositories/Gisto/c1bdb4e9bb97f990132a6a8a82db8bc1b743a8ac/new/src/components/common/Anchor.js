import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Link = styled.a`
  cursor: pointer;
  color: ${(props) => props.color ? props.color : props.theme.baseAppColor};
  text-decoration: none;

  :hover {
    color: ${(props) => props.theme.headerBgLightest};
  }
`;

const Anchor = ({
  href, onClick, children, download, color, className
}) => (
  <Link href={ href }
        className={ className }
        color={ color }
        download={ download }
        onClick={ onClick }>
    { children }
  </Link>
);

Anchor.propTypes = {
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
  download: PropTypes.string,
  color: PropTypes.string,
  className: PropTypes.string
};

export default Anchor;
