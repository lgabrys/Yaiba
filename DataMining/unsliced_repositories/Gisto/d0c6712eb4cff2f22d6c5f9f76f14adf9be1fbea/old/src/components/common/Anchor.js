import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { baseAppColor, headerBgLightest } from 'constants/colors';

const Link = styled.a`
  cursor: pointer;
  color: ${baseAppColor};
  text-decoration: none;
  
  :hover {
    color: ${headerBgLightest};
  }
`;

const Anchor = ({
  href, onClick, children, download
}) => (
  <Link href={ href }
        download={ download }
        onClick={ onClick }>
    { children }
  </Link>
);

Anchor.propTypes = {
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
  download: PropTypes.string
};

export default Anchor;
