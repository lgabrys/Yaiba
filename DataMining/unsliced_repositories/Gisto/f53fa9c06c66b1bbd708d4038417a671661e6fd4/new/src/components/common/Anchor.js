import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { baseAppColor } from 'constants/colors';

const Link = styled.a`
  cursor: pointer;
  color: ${baseAppColor};
  text-decoration: none;
`;

const Anchor = ({
  key, href, onClick, children, download
}) => (
  <Link href={ href }
        key={ key }
        download={ download }
        onClick={ onClick }>
    { children }
  </Link>
);

Anchor.propTypes = {
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
  key: PropTypes.any,
  download: PropTypes.string
};

export default Anchor;
