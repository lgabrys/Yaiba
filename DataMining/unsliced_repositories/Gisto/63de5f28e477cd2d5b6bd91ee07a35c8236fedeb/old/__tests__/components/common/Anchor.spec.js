import Anchor from 'components/common/Anchor';

const setup = (props) => shallow(<Anchor { ...props }>link</Anchor>);

describe('COMPONENTS - <Anchor>', () => {
  test('render Anchor', () => {
    const props = {
      download: 'file.zip',
      href: 'https://github.com'
    };
    const component = setup(props);

    expect(component).toMatchSnapshot();
  });

  test('handle click', () => {
    const props = {
      download: 'file.zip',
      href: 'https://github.com',
      onClick: jest.fn()
    };
    const component = setup(props);

    component.simulate('click');

    expect(props.onClick).toHaveBeenCalled();
  });
});
