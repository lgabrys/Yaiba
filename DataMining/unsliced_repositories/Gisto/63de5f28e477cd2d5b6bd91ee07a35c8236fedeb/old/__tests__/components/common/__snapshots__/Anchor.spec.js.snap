// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS - <Anchor> render Anchor 1`] = `
<Link
  download="file.zip"
  href="https://github.com"
>
  link
</Link>
`;
